File: Day 9 : Recursion 3/Solution.java
import java.util.*;

class Result
{
    public static int factorial(int n)
    {
        if(n > 1)
        {
            return n * factorial(n-1);
        }
        else
        {
            return 1;
        }
    }
}

public class Solution
{
    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int result = Result.factorial(n);
        System.out.println(result);
        scan.close();
    }
}

File: Day 9 : Recursion 3/Solution.ts
function factorial(n: number): number {
    if (n === 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

function main() {
    const n: number = parseInt(readLine());
    const result: number = factorial(n);
    console.log(result);
}